deepmd/pt/optimizer/hybrid_muon.py

```
@@ -880,6 +1306,165 @@ def _get_ns_buffers(
+        for (rows, cols, dev, dt), bucket_entries in gram_buckets.items():
+            min_dim = min(rows, cols)
+            max_dim = max(rows, cols)
+            transposed = rows > cols
+            sb_key = (min_dim, dev, dt)
+            if sb_key not in super_buckets:
+                super_buckets[sb_key] = []
+            super_buckets[sb_key].append((rows, cols, transposed, bucket_entries))
+
+        gram_orth = self._get_gram_orthogonalizer()
+
+        for (min_dim, _dev, _dt), sub_list in super_buckets.items():
```

Ruff (0.15.10)

[warning] Loop control variable `min_dim` not used within loop body

(B007)